chromadb/utils/embedding_functions.py

```diff
@@ -244,9 +248,13 @@ class ONNXMiniLM_L6_V2(EmbeddingFunction):
 
     # https://github.com/python/mypy/issues/7291 mypy makes you type the constructor if
     # no args
-    def __init__(self) -> None:
+    def __init__(self, preferred_providers: Optional[List[str]] = None) -> None:
         # Import dependencies on demand to mirror other embedding functions. This
         # breaks typechecking, thus the ignores.
+        # convert the list to set for unique values
+        self._preferred_providers = (
```

Comment:
I think the order of the providers is important no? So converting to a set could be destructive?

Reply:
" Kernels from different execution providers are chosen in the priority order given in the list of providers. "

https://onnxruntime.ai/docs/api/python/api_summary.html

Reply:
Agree. My concern was validating user input (e.g. avoiding duplicates)